parallel/task.go

```diff
@@ -212,7 +212,7 @@ func (p *ParallelTasks) Do() *ParallelTasks {
 
 	for i, task := range p.tasks {
 		if !p.waitThreshold() {
```

Comment:
This is where channel initialized. Should ensure they will be release in this func.